src/NzbDrone.Common/Cache/Cached.cs

```diff
@@ -42,7 +43,13 @@ public void Set(string key, T value, TimeSpan? lifetime = null)
 
             if (lifetime != null)
             {
-                System.Threading.Tasks.Task.Delay(lifetime.Value).ContinueWith(t => _store.TryRemove(key, out var temp));
+                System.Threading.Tasks.Task.Delay(lifetime.Value).ContinueWith(t =>
+                {
+                    if (_store.TryRemove(key, out var temp2) && !temp2.IsExpired())
```

Comment:
Would it be better to try to retrieve the key, and only remove if expired, rather than removing and re-adding if it wasn't actually expired?